pennylane/decomposition/controlled_decomposition.py

```diff
@@ -221,7 +221,12 @@ def controlled_x_decomp(*_, wires, control_wires, control_values, work_wires, **
 
 @functools.lru_cache()
 def base_to_custom_ctrl_op():
-    """A dictionary mapping base op types to their custom controlled versions."""
+    """A dictionary mapping base op types to their custom controlled versions.
+
+    This dictionary is used under the assumption that all custom controlled operations do not
+    have resource params (which is why `ControlledQubitUnitary` is not included here).
```

Comment:
Is there a plan to generalize this? Are there other ops than `ControlledQubitUnitary` affected by this? :thinking:

Reply:
I think `ControlledQubitUnitary` is the only exception here.